content/post/partiels-fonction-hugo.md

```diff
@@ -0,0 +1,346 @@
+2. `site.Taxonomies.tags` retourne une collection de tous les tags du site avec leur objet `.Page`. Le _point_ représente le contexte de notre partiel, ici le nom de notre tag, que nous transformons en URL pour correspondre à sa clef dans `site.Taxonomies.tags`.
+3. Nous avons de fortes chances de tomber sur une  `.Page`, mais `with` ajoute une vérification supplémentaire et nous permet en plus de changer de contexte.
+4. Nous stockons le tag de la page dans notre _variable retournée_.
+5. 🎉
+
+**👍 Beau boulot !**
+
+Et pour les autres taxonomies comme les `categories`? Hors de question de copier/coller tout cela dans un nouveau partiel pour remplacer `site.Taxonomies.tags` par `site.Taxonomies.categories`. 🙅‍♀️
+
+Nous voulons pouvoir écrire ceci :
+
+```go-html-template
+{{ range .Params.categories }}
+  {{ with partial "func/GetTermPage" (dict "taxonomy" "categories" "term" .) }}
+    <a href="{{ .RelPermalink }}">{{ .Title }}</a>
+  {{ end }}
+{{ end }}
+```
+
+## Gestion des arguments
+
+Jusqu'à maintenant nous n'avons passé qu'un seul argument à nos partiels de fonction, où le contexte ne contenait qu'un élément. Mais ici, il nous en faut deux : la taxonomie et son terme. Notre contexte devra donc être un tableau associatif qui contiendra les deux.
+
+Nous mettons donc notre partiel à jour :
+
+```go-html-template
+{{ $return := false }}
+
+{{/* 1. */}}
+{{ $taxonomy := "tags" }}
+{{ with .taxonomy }}
+  {{ $taxonomy = . }}
+{{ end }}
+
+{{/* 2. */}}
+{{ with $term := .term }}
+  {{ with index site.Taxonomies $taxonomy }}
+    {{ with index . (urlize $term) }}
+      {{ with .Page }}
+        {{ $return = . }}
+      {{ end }}
+    {{ end }}
+  {{ end }}
+{{ end }}
+
+{{/* 3. */}}
+{{ return $return }}
+```
+
+1. Maintenant que nous passons un argument `term`, nommer notre variable retournée `$term` pourrait prêter à confusion. Appelons la maintenant `$return`  pour bien marquer que c'est **la** _variable retournée_.
+2. Si aucun argument `.term` n'est présent, la _variable retournée_ devrait rester vide. Avant d'aller plus loin, nous faisons appel à `with` pour nous assurer que `.term` est bien défini et nous stockons cette valeur initiale afin de pouvoir y accéder quelque que soit notre contexte. Ces quelques lignes sont d'ailleurs une très bonne illustration de changements de contexte !
+3. 🎉
+
+## Mise en cache !
+
+OK très bien, mais je veux une fonction qui liste les tags d'une page et qui me renvoie un tableau de tableaux associatifs qui contiennent chacun des données structurées comme `.URL` et `.Name`. De cette façon, si je veux passer de `.RelPermalink` à `.Permalink` dans le futur, je peux le faire dans ma _variable retournée_ plutôt que dans chaque fichier de modèle où je souhaite afficher ces liens.
+
+C'est l'occasion idéale pour voir comment appeler une _variable retournée_ depuis une _variable retournée_ et mettre en cache sa valeur. :sweat_smile:
```

Comment:
L'original: _It’ll give us a great opportunity to call a returning partial from within a returning partial and cache its value._. Donc c'est plus l'occasion d'appeler une fonction de partiel depuis une fonction de partiel...

Reply:
oulà oui, ça voulait rien dire sinon 😊